orc/src/main/java/org/apache/iceberg/orc/ORC.java

```diff
@@ -162,11 +162,9 @@ private WriteBuilder createContextFunc(Function<Map<String, String>, Context> ne
     public <D> FileAppender<D> build() {
       Preconditions.checkNotNull(schema, "Schema is required");
 
-      Configuration conf;
+      Configuration conf = new Configuration();
       if (file instanceof HadoopOutputFile) {
-        conf = ((HadoopOutputFile) file).getConf();
-      } else {
-        conf = new Configuration();
+        ((HadoopOutputFile) file).getConf().forEach(entry -> conf.set(entry.getKey(), entry.getValue()));
```

Comment:
Seems we will merge the hadoop file's configuration with the default environment configuration by default if it's a `HadoopOutputFile`, right ? 

Should we just use the original code piece ( according to [here](https://github.com/apache/iceberg/pull/4291/files#diff-34d7fce4c1d9417fa9342247cf5ace0636cd86591efbed0555ae80197f79303dL90-L94)):

```java
      if (file instanceof HadoopOutputFile) {
        this.conf = new Configuration(((HadoopOutputFile) file).getConf());
      } else {
        this.conf = new Configuration();
      }
```

Reply:
makes sense, made the changes